### PR TITLE
Specify vehicle part repair requirements in JSON

### PR DESCRIPTION
**Pull request by mugling**
### Significant changes
- Sets groundwork for sensible vehicle repairs - no more welding timber or duct taping engine blocks. No significant balance changes are however included in this PR. Supports non-repairable parts although again as yet no implemented cases.
- Further ~~exorcism~~ refactoring of `veh_interact` class. In particular the class is now encapsulated and returns a `player_activity`. Refactor some duplicate code in to static functions
- Further fixes for skills other than `mechanics` (follows #18429). Time bonus for installation/removal is now dependent upon relevant skills (as opposed to hardcoded against `mechanics`). For a single skill requirement you get 10% bonus per excess level, for two skills 5% per level etc to a maximum of 50%.
- Vehicle damage display is now handled via discrete levels (similar to items) and uses the same colors. Repair requirements are also specified per damage level. Considerable improvement to gameplay as running over a shrub no longer prompts to repair (although running over a lot might due to the cumulative effect).
- Demonstration item is cloth roof (repaired using tailoring skill, sewing tool and fabric

### CODE REVIEW

**Comment by mugling** [comment]
Ready. Note updated preamble

**Comment by mugling** [comment]
@illi-kun if your lurking could you try a merge test on this? I'm blocked on all my other PR's so it would be good to have an open route for progress and the vehicle code is the obvious choice

**Comment by illi-kun** [comment]
Sorry, I'm out of PC already. I'll take a look on this tonight if no one beats me on this.

File: data/json/requirements/toolsets.json [hunk 1]
```diff
@@ -15,6 +15,12 @@
     "tools": [ [ [ "forge", 20 ], [ "char_forge", 20 ], [ "oxy_torch", 20 ] ] ]
   },
   {
+    "id": "sewing_standard",
+    "type": "requirement",
+    "//": "Crafting or repair of fabric items",
+    "tools": [ [ [ "sewing_kit", 1 ], [ "tailors_kit", 1 ], [ "needle_wood", 1 ] ] ]
```
Comment (sethsimon) [inline data/json/requirements/toolsets.json]: Why not the bone needle?

Comment (Coolthulhu) [inline data/json/requirements/toolsets.json]: Also why specify wooden needle and tailor kit? Doesn't the system check for subs anyway?

Comment (cainiaowu) [inline data/json/requirements/toolsets.json]: Or make a "SEWING" quality that can consume charge.
The idea of tool quality is not quite heavily used.

Comment (mugling) [inline data/json/requirements/toolsets.json]: > Why not the bone needle?

Omission. Example of why I want to use tool sets

> Also why specify wooden needle and tailor kit? Doesn't the system check for subs anyway?

Existing bugs with this approach #17517

> Or make a "SEWING" quality that can consume charge.

Same issue as `WELDING`. Long term goal. Will necessitate further rewrite of crafitng.

**Comment by mugling** [comment]
Qualities consuming charges is a long-term goal (its an excellent idea that has been floated before) but supporting it is trivial. Anything else left to check on this or could anyone lurking try for a merge run so I can continue with cleanup of `veh_interact`

**Pull request merged**